### ISSUE TITLE
Add additional configuration options for client-side routing announcements

### ISSUE THREAD

**Issue by madalynrose** [gatsbyjs/gatsby#21059]
## Summary

Add additional configuration options for the `RouteAnnouncer` created in #19290 by adding a `<RouteAnnouncement>` Gatsby component.

### Basic example

```jsx
<RouteAnnouncement visuallyHidden={true}>
  {blogPost.title}
</RouteAnnouncement>
```

or

```jsx
<RouteAnnouncement>
	<h1>Blog Post Title</h1>
</RouteAnnouncement>
```

### Motivation

Allowing developers to specify what exactly should be announced instead of inferring that information based on the `h1` or `title` of a page will ensure route change announcements are accurate. This could be achieved by creating a Gatsby `RouteAnnouncement` component that can wrap relevant content, as proposed in this [comment](https://github.com/gatsbyjs/gatsby/pull/19290#issuecomment-550067795). This component could also have error checking to make sure announcements aren't too long and have an option for being visually hidden, since they are intended to be consumed by assistive technologies.
